src/components/Buttons/docs/Button.md

```diff
@@ -233,3 +233,23 @@ import { Button, Icon } from 'ic-snacks'
     </span>
   </div>
 ```
+
+### Custom Element
+Buttons can render a custom base HTML Element or React Component, while leveraging the full styling of a Snacks Button.
+
+```jsx
+import { Button, Icon, Link } from 'ic-snacks';
+
+  <div style={{padding: '24px'}}>
+    <span style={{marginRight: '24px'}}>
+      <Button snacksStyle="primary" size="small" customElement='span'>
+        Using a Custom HTML Element
+      </Button>
+    </span>
+    <span style={{marginRight: '24px'}}>
+      <Button snacksStyle="primary" size="small" customElement={Link} customElementProps={{ href: '/foo' }}>
```

Comment:
This isn't an example we should use since `Button` supports this use case already via the `href` prop.

```js
<Button href="/foo">Foo</Button>
```